miniscope_io/stream_daq.py

```diff
@@ -237,7 +243,17 @@ def _fpga_recv(
         if self.config.LSB:
             pre = pre[::-1]
         while True:
-            buf = dev.readData(read_length)
+            try:
+                buf = dev.readData(read_length)
+            except (EndOfRecordingException, StreamReadError):
+                self.terminate.value = True
+                break
+
+            if self.config.mode == 'RAW_RECORD':
```

Comment:
flagging this as the other place we would need to modify if we went with suggestion ^